### ISSUE TITLE
Lazy initialization of library breaks click on iPhone

### ISSUE THREAD

**Issue by mihalikv** [leongersen/noUiSlider#1246]
## Expected Behavior
When I click on "Go to google", browser should redirect to the Google.com.

## Current Behavior
When I click on "Go to google", I am still on same page.

## Steps to Reproduce
https://jsfiddle.net/v7axdwLm/ -> Click on link on iPhone

## Context (Environment)
Tested on real device and also in browserstack 
![image](https://github.com/leongersen/noUiSlider/assets/10236315/4b13baef-e507-4740-9a17-28583e028cf0)

## Possible Implementation
I tried to change code of adding event listeners in various way and this change helped but it definitely break rest of the lib
![image](https://github.com/leongersen/noUiSlider/assets/10236315/f4fee2e3-f538-4768-9f78-613dc3c1dd51)

**Comment by leongersen** [comment]
The lazy init seems to work fine for me, if you click elsewhere in the document.

The reason the Jsfiddle does not navigate is unrelated:

```
Refused to display 'https://www.google.com/' in a frame because it set 'X-Frame-Options' to 'SAMEORIGIN'.
```

**Comment by mihalikv** [comment]
I changed jsfiddle the link in issue https://jsfiddle.net/v7axdwLm/2/

**Comment by mihalikv** [comment]
https://github.com/leongersen/noUiSlider/assets/10236315/d8d0efb5-cc66-41a8-8c5a-f6845bcf2c3e

Recording from real iPhone.

**Comment by leongersen** [comment]
I don't think that's related to noUiSlider.

**Comment by mihalikv** [comment]
Yeah it's not related to noUiSlider, it's more like noUiSlider is affected by some of the conventions of safari/iPhone. When you add an element dynamically and then attach the event to it, it will affect clicking anywhere else on the page.

Here is minimal reproduction without noUIslider: https://jsfiddle.net/1ktyhfev/10/
and here is the "fixed" version with setTimeout: https://jsfiddle.net/1ktyhfev/11/

I test it also with "fix" in noUiSlider and it works for me 
![image](https://github.com/leongersen/noUiSlider/assets/10236315/e632efb6-afc7-4446-9db7-16b06d149dbe)

Unfortunately, I cannot find any related articles about this kind of bug.
